cloudflared/rootfs/etc/cont-init.d/cloudflared-config.sh

```diff
@@ -354,6 +393,10 @@ main() {
         createTunnel
     fi
 
+    if hasCustomConfig ; then
```

Comment:
I think we need to adapt this logic. If the login and the tunnel creation are done in our add-on and the config should be created in a custom way, there is no way for the user to know the tunnel UUID to put into the config in the first place. Do you know, what I mean?

So we have different possible situations:

1. hasCustomConfig and tunnel_file and cert are existing (all put there by user): fine, works already
2. !hasCustomConfig: fine, works already
2. hasCustomConfig and tunnel file is missing: create tunnel (and previously create cert if not there). Then the only way for it to work would be to automatically add the created tunnel UUID and the link to the credentials file in the custom_config. This would be quite nice but needs to be stated clearly in the docs.md. The other option here would be to not allow that combination and make the user add everything himself if he wants a custom config.

Does that make sense or am I missing something?

Reply:
The proposed logic will use the created cert and tunnel config from already implemented logic. 

1. Exactly: it will work if put there by user manually or created automatically with add-on logic
2. yep
3. In my opinion, the logic of custom config should only be used by advanced users anyway. Also, the config can only be created if the corresponding contents of the Cloudflared documentation are known. Thus, it should also be possible for the user to manually read the UUID from the tunnel.json and add it. Especially because all required files are located in the same directory. Don't you think we complicate the general add-on logic if we try to catch all cases? I wouldn't like to partially auto-complete a custom config with the add-on configuration.